src/seedu/addressbook/AddressBook.java

```diff
@@ -373,11 +380,71 @@ public static String executeCommand(String userInputString) {
         }
     }
 
+    /**
+     * Given the last displayed index, delete the person from address book
+     * and adds the new person
+     * 
+     * @return feedback display message for the operation result
+     */
     private static String executeEdit(String commandArgs) {
-    	final Optional<String[]> decodeResult = decodePersonFromString(commandArgs);
-    	final String[] personToAdd = decodeResult.get();
+        String message = checkIfEditUserExists(commandArgs);
+        if(message.equals(MESSAGE_INPUT_NEW_DETAILS)){
+            String newDetails = inputNewDetails(message);
+            return addEditPerson(newDetails);
+        }
```

Comment:
ta! daaaa! coding standard violation ;-) 
need to have the `else` "cuddled" between the braces `} else {`